includes/admin/class-wc-stripe-admin-notices.php

```diff
@@ -458,6 +463,47 @@ public function payment_methods_check_environment() {
 		}
 	}
 
+	/**
+	 * Adds a notice to the order details page if the subscription payment method has been detached.
+	 *
+	 * @return void
+	 */
+	public function subscription_check_detachment() {
+		if ( ! WC_Stripe_Page_Helper::is_subscription_edit_page() ) {
+			return;
+		}
+
+		global $theorder;
+
+		// If $theorder is empty (i.e. non-HPOS), fallback to using the global post object.
+		if ( empty( $theorder ) && ! empty( $GLOBALS['post']->ID ) ) {
+			$theorder = wcs_get_subscription( $GLOBALS['post']->ID );
+		}
+
+		if ( ! isset( $theorder ) || ! $theorder instanceof WC_Subscription ) {
+			return;
+		}
+
+		$subscription = $theorder;
+		if ( WC_Stripe_Subscriptions_Helper::is_subscription_payment_method_detached( $subscription ) ) {
+			$detached_message  = __( 'The payment method for this subscription has been detached, <strong>preventing renewals</strong>. Share the payment method page link with the customer to update it or manually set the Stripe Payment Method ID meta field in the subscriptions details\' "Billing" section to another from the customer\'s page on Stripe. Below are the last subscriptions affected and the links as mentioned earlier:<br />', 'woocommerce-gateway-stripe' );
```

Comment:
Same suggestion as the [other PR](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4391#discussion_r2129111891). 
Additionally, on this subscription edit page, we are only talking about this specific subscription. `Below are the last subscriptions affected and the links as mentioned earlier:` This part talks about more than one subscriptions which is a bit off in this context.

Reply:
Good point. Updated it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4393/commits/205da17f04c94aee2c4dd8eb5122a25090a90799 and https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4393/commits/941bda2a9748de412ebbcd490a094a34f3fde0ac